### PR TITLE
auto-moderator tweaks: pass along record URI, create report for takedown action

### PR DESCRIPTION
**Pull request by bnewbold**
Some polish on the auto-takedown process:

- pass record URI along to abyss for easier tracking/reporting
- log that a hard takedown is happening ("warn" log level)
- create a report, in addition to the takedown itself, as a paper trail for mod team

### CODE REVIEW

**Comment by bnewbold** [comment]
I fixed a couple issues, but can't figure out why the two takedown tests are still failing (mod action not getting created). To run locally, in bsky dir:

```sh
../dev-infra/with-test-redis-and-db.sh yarn run jest tests/auto-moderator/takedowns.test.ts
```

File: packages/bsky/src/auto-moderator/abyss.ts [hunk 1]
```diff
@@ -69,8 +74,8 @@ export class Abyss implements ImageFlagger {
     return labels
   }
 
-  getReqUrl(params: { did: string }) {
-    return `${this.endpoint}/xrpc/com.atproto.unspecced.scanBlob?did=${params.did}`
+  getReqUrl(params: { did: string; uri: string }) {
+    return `${this.endpoint}/xrpc/com.atproto.unspecced.scanBlob?did=${params.did}&uri=${params.uri}`
```
Comment (devinivy) [inline packages/bsky/src/auto-moderator/abyss.ts:78]: The `did` param should be safe, but I think we'll want to encode the `uri` param with `encodeURIComponent()`.

Comment (devinivy) [reply packages/bsky/src/auto-moderator/abyss.ts:78]: Could also do something along these lines:
```ts
const search = new URLSearchParams(params)
return `${this.endpoint}/xrpc/com.atproto.unspecced.scanBlob?${search.toString()}`
```

**Comment by devinivy** [comment]
@bnewbold I took a look at the error, and I think the issue is that some auth needs to be passed along for `createReport`.

<details>
  <summary> 👈 error log details in here</summary>

   ```
[14:28:11.316] ERROR (AuthMissing/55694): error in xrpc method com.atproto.moderation.createReport
    status: 401
    message: ""
[14:28:11.317] ERROR (bsky:labeler/55694): failed to persist takedown
    uri: "at://did:plc:gy6nqo27cluaibqjdbt6qvrd/app.bsky.feed.post/3k7vqk2q4cc2x"
    imgCids: [
      "bafkreigy5p3xxceipk2o6nqtnugpft26ol6yleqhboqziino7axvdngtci",
      "bafkreiaivizp4xldojmmpuzmiu75cmea7nq56dnntnuhzhsjcb63aou5ei"
    ]
    results: [
      [],
      [
        "kill-it"
      ]
    ]
    err: {
      "type": "XRPCError",
      "message": "Authentication Required",
      "stack":
          Error: Authentication Required
              at ServiceClient.call (/Users/devinivy/Documents/bluesky/atproto/packages/xrpc/src/client.ts:126:15)
              at processTicksAndRejections (node:internal/process/task_queues:95:5)
              at AutoModerator.persistTakedown (/Users/devinivy/Documents/bluesky/atproto/packages/bsky/src/auto-moderator/index.ts:226:7)
              at AutoModerator.checkImgForTakedown (/Users/devinivy/Documents/bluesky/atproto/packages/bsky/src/auto-moderator/index.ts:188:7)
              at async Promise.all (index 2)
              at /Users/devinivy/Documents/bluesky/atproto/packages/bsky/src/auto-moderator/index.ts:96:7
              at run (/Users/devinivy/Documents/bluesky/atproto/node_modules/.pnpm/p-queue@6.6.2/node_modules/p-queue/dist/index.js:163:29)
      "status": 401,
      "error": "AuthMissing",
      "success": false,
      "headers": {
        "access-control-allow-origin": "*",
        "connection": "keep-alive",
        "content-length": "59",
        "content-type": "application/json; charset=utf-8",
        "date": "Thu, 21 Sep 2023 12:28:11 GMT",
        "etag": "W/\"3b-MJ9BSQx7lf7MrSwUs76YjltpuIQ\"",
        "keep-alive": "timeout=90",
        "vary": "Accept-Encoding",
        "x-powered-by": "Express"
      }
    }
   ```
</details>

**Comment by bnewbold** [comment]
ah, thanks, hadn't noticed `test.log`, and have been kind of just brute-forcing this without thinking it through. the indexing service has admin auth for the PDS to do hard takedown, and does have an identity (DID), but that identity probably isn't registered/valid on the PDS for auth.

I updated to, I think, create the report locally instead of on PDS, regardless of pushAgent existence. I'm groping a little blind here (why is it "pushAgent", and separately why is it `modSrvc.logAction` not `createAction`?), but maybe this fixes things.

**Review by dholms — APPROVED** [review]
yup looks good 👍 

just the uri encoding thing that devin flagged

**Comment by dholms** [comment]
> the indexing service has admin auth for the PDS to do hard takedown, and does have an identity (DID), but that identity probably isn't registered/valid on the PDS for auth.

yeah a lot of this stuff is in kinda a weird state with the moderation transition - creating the report in just the appview is fine (we don't use pds reports anymore) 

> why is it "pushAgent"

we treat appview moderation state as the canoncial store, but during the transition period, we were reflecting it in both appview & PDS and actions hit the PDS first. "pushAgent" because it's "the agent that pushes actions to the pds" (tho i agree we could rework the naming of this stuff a bit)

>  and separately why is it modSrvc.logAction not createAction?

tbh i'm not sure on that one. but I think the idea is that the action doesn't actually "do" anything. so we log the existence of an action (for instance a takedown) then we apply the action (resolving reports, doing takedowns etc).

**Comment by bnewbold** [comment]
thank you both for the review on my messy code and walking me through this! learning just enough typescript to be dangerous

**Pull request merged**